dwave/plugins/torch/boltzmann_machine.py

```diff
@@ -26,103 +26,176 @@
+        self._quadratic = torch.nn.Parameter(5.0 * (2 * torch.rand(self._edges) - 1))
+
+        self.register_buffer("_edge_idx_i", edge_idx_i)
+        self.register_buffer("_edge_idx_j", edge_idx_j)
+
+        self._setup_hidden(nodes, edges, hidden_nodes)
+
+    @property
+    def linear(self):
+        return self._linear
+
+    @property
+    def quadratic(self):
+        return self._quadratic
+
+    @property
+    def nodes(self):
+        return self._nodes
+
+    @property
+    def var_to_idx(self):
+        return self._var_to_idx
+
+    @property
+    def idx_to_var(self):
+        return self._idx_to_var
+
+    @property
+    def n_nodes(self):
+        return self._n_nodes
+
+    @property
+    def n_edges(self):
+        return self._n_edges
+
+    @property
+    def visible_idx(self):
+        return self._visible_idx
+
+    @property
+    def hidden_idx(self):
+        return self._hidden_idx
+
+    @property
+    def fully_visible(self):
+        return self._fully_visible
+
+    @property
+    def has_hidden(self):
+        return not self.fully_visible
+
+    @property
+    def edge_idx_i(self):
+        return self._edge_idx_i
 
-        self.m = len(edges)
-        edge_idx_i = torch.tensor([self.var_to_idx[i] for i, j in edges])
-        edge_idx_j = torch.tensor([self.var_to_idx[j] for i, j in edges])
+    @property
+    def edge_idx_j(self):
+        return self._edge_idx_j
 
-        self.h = torch.nn.Parameter(0.05 * (2 * torch.rand(self.n) - 1))
-        self.J = torch.nn.Parameter(5.0 * (2 * torch.rand(self.m) - 1))
+    @property
+    def theta(self) -> torch.Tensor:
+        """Parameters of the model---linear and quadratic biases---as a one-dimensional
+        tensor. The linear and quadratic biases are concatenated in the order as defined
+        by the model's input ``nodes`` and ``edges``."""
+        return torch.cat([self._linear, self._quadratic])
 
-        self.register_buffer("edge_idx_i", edge_idx_i)
-        self.register_buffer("edge_idx_j", edge_idx_j)
+    def _setup_hidden(self, nodes: Iterable[Hashable], edges: Iterable[tuple[Hashable, Hashable]],
+                      hidden_nodes: Iterable[Hashable] = None):
+        """Preprocess some indexes to enable vectorized computation of effective fields of hidden units.
 
-        self.fully_visible = hiddens is None
-        if not self.fully_visible:
-            connected_hidden = any(a in hiddens and b in hiddens for a, b in edges)
+        Args:
+            nodes (Iterable[Hashable]): Nodes of the model.
+            edges (Iterable[tuple[Hashable, Hashable]]): Edges of the model.
+            hidden_nodes (Iterable[Hashable], optional): Hidden nodes of the model. Each hidden node
+                should also be in ``nodes``.
+        """
+        self._fully_visible = hidden_nodes is None
+        if not self._fully_visible:
+            hidden_nodes = list(hidden_nodes)
```

Comment:
Any reason not to have this be an attribute as with nodes and edges?

Reply:
It should 😝. Added!